src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -1390,66 +1390,83 @@ public final Response deleteBooking(@Context final HttpServletRequest request,
    *
    * @param request  so we can determine if the user is logged in
    * @param eventId  event booking containing updates, must contain primary id.
-   * @param userId   the user to be promoted.
+   * @param userIds   the user IDs to be promoted.
    * @param attended boolean value representing whether the user was present, true, or absent, false.
    * @return the updated booking.
    */
   @POST
-  @Path("{event_id}/bookings/{user_id}/record_attendance")
+  @Path("{event_id}/bookings/record_attendance")
   @Produces(MediaType.APPLICATION_JSON)
   @GZIP
-  @Operation(summary = "Update the attendance status of a user for an event.")
+  @Operation(summary = "Update the attendance status of users for an event.")
   public final Response recordEventAttendance(@Context final HttpServletRequest request,
                                               @PathParam("event_id") final String eventId,
-                                              @PathParam("user_id") final Long userId,
+                                              final List<Long> userIds,
                                               @QueryParam("attended") final Boolean attended) {
+    List<Long> failedUserIds = new ArrayList<>();
+
+    if (userIds.isEmpty()) {
+      return new SegueErrorResponse(Status.BAD_REQUEST, "No User Ids provided").toResponse();
+    }
+
     try {
       RegisteredUserDTO currentUser = this.userManager.getCurrentRegisteredUser(request);
-      RegisteredUserDTO userOfInterest = this.userManager.getUserDTOById(userId);
       IsaacEventPageDTO event = this.getAugmentedEventDTOById(request, eventId);
 
       if (!bookingManager.isUserAbleToManageEvent(currentUser, event)) {
         return SegueErrorResponse.getIncorrectRoleResponse();
       }
 
-      EventBookingDTO eventBookingDTO = this.bookingManager.recordAttendance(event, userOfInterest, attended);
-      this.getLogManager().logEvent(currentUser, request,
-          SegueServerLogType.ADMIN_EVENT_ATTENDANCE_RECORDED,
-          Map.of(
-              EVENT_ID_FKEY_FIELDNAME, event.getId(),
-              USER_ID_FKEY_FIELDNAME, userId,
-              ATTENDED_FIELDNAME, attended,
-              EVENT_DATE_FIELDNAME, event.getDate(),
-              EVENT_TAGS_FIELDNAME, event.getTags()
-          ));
-
-      if (event.getTags().contains("teacher")) {
-        this.userBadgeManager.updateBadge(userOfInterest,
-            UserBadgeManager.Badge.TEACHER_CPD_EVENTS_ATTENDED, eventId);
+      for (Long userId : userIds) {
+        recordAttendanceForUser(userId, event, attended, failedUserIds, currentUser, request);
       }
 
-      return Response.ok(eventBookingDTO).build();
+      if (failedUserIds.isEmpty()) {
+        return Response.ok().build();
+      } else {
+        String errorMessage =
+            String.format("One or more bookings could not be updated: %s", failedUserIds);
+        return new SegueErrorResponse(Status.BAD_REQUEST, errorMessage).toResponse();
```

Comment:
what's the outcome if one or more can't be updated? Does it ditch all changes or just complete the ones it can? If the latter this is probably a warning rather than an error

Reply:
It's constructed so that I doesn't error out midway through. For example if there are 10 userIds in the list it should try each one, and add any failed UserIDs to a list.  When it has finished, it returns that error if there are any IDs in the list.  I based this on how I did it for `modifyUsersTeacherPendingStatus`. 

I'd expect this mostly _shouldn't_ happen anyway as requests will happen after selecting users with checkboxes on the front end, so non-existent userIds or bookings shouldn't be possible.

Reply:
What happens on the frontend? It's a 'partial success' situation, for which there is nothing official in the http standards (207 is nearest). Is the glass half full or half empty? Is it a partial success or a partial fail? If you handle error states on the UI by displaying an error box, then you might not want to return a 500 when some things worked.

Reply:
500 errors cause a big horrible front end full page error (like to avoid this unless something has gone very wrong).  Errors under 500 give a small error popup with additional information, so returning BAD_REQUEST here (400) has the desired experience for the user. I'd consider bad request to be an accurate description also, as it shouldn't be possible for them to send a request with nonexistent IDs or bookings - if it happens, they've done something weird!